### DISCUSSION TITLE
Issue with @shadcn/ui ThemeProvider during pnpm build

### DISCUSSION THREAD

**Discussion by marufh1** [shadcn-ui/ui#3295] [General]
Overview: While using the ThemeProvider from @shadcn/ui during development with pnpm dev, everything works as expected. However, when building the project using pnpm build, it encounters issues.

Reproduction Steps:

-Start the development server using pnpm dev.
-Observe that the application functions correctly without errors.
-Build the project using pnpm build.
-Observe that the build process encounters errors related to the ThemeProvider.

-Import RootLayout and ThemeProvider.
-Use ThemeProvider inside RootLayout with specific props.
-TypeScript raises an error indicating 'NextThemesProvider' cannot be used as a JSX component.

**Expected Behavior:** The application should build successfully without errors using pnpm build, just like it does during development with pnpm dev. TypeScript should recognize 'NextThemesProvider' as a valid JSX component and not raise an error.
**Actual Behavior:** During the build process with pnpm build, the application encounters errors related to the ThemeProvider from @shadcn/ui. These errors prevent the successful completion of the build process. TypeScript raises an error: 'NextThemesProvider' cannot be used as a JSX component.

```
Error Messages: Type error: 'NextThemesProvider' cannot be used as a JSX component.
  Its return type 'string | number | boolean | Element | Iterable<ReactNode>' is not a valid JSX element.
    Type 'string' is not assignable to type 'ReactElement<any, any>'.

  5 |
  6 | export function ThemeProvider({ children, ...props }: ThemeProviderProps) {
> 7 |   return <NextThemesProvider {...props}>{children}</NextThemesProvider>;
    |           ^
  8 | }
```

### Code Samples:
**RootLayout.tsx:**
```
export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang='en' suppressHydrationWarning>
      <body className={`container ${inter.className}`}>
        <ThemeProvider
          enableSystem // Enable system theme detection
          attribute='class' // Apply theme class to the body element
          defaultTheme='system' // Use system theme as default
          disableTransitionOnChange // Prevent theme transitions on change
          themes={['light', 'dark', 'orange']} // Define available themes
        >
          {children}
        </ThemeProvider>
      </body>
    </html>
  );
}
```

**theme-provider.tsx**
```
"use client";

import * as React from "react";
import { ThemeProvider as NextThemesProvider } from "next-themes";
import { type ThemeProviderProps } from "next-themes/dist/types";

export function ThemeProvider({ children, ...props }: ThemeProviderProps) {
  return <NextThemesProvider {...props}>{children}</NextThemesProvider>;
}
```

TypeScript version: 4.9.5
React version: 18.2.0

**Comment by marufh1** [comment]
**Resolved issue by downgrading the next-themes version from 0.3.0 to 0.2.1.**

  Explanation:
  Previously, when using next-themes version 0.3.0, encountered errors during the build process. These errors caused the build to fail without providing clear output or error messages. By downgrading to version 0.2.1, the build process completed successfully without encountering any errors.

  Steps taken:
  1. Identified the next-themes package as a potential source of the issue.
  2. Reviewed the changelog and release notes for next-themes to understand any changes or updates between versions.
  3. Decided to downgrade to version 0.2.1 as a workaround to resolve the build errors.
  4. Updated the package.json file to specify next-themes version 0.2.1.
  5. Ran the build process again to confirm the resolution.

Outcome:
  By downgrading the next-themes package to version 0.2.1, the build process completed successfully without errors. The application functions as expected, and there are no apparent issues with the downgraded version.

**Reply by Manuelentrena** [reply]
Thanks, I have had the same error during 2 hours.  I have tried another way to solve the error but without success. Downgrading to version 0.2.1 resolves build errors for now.
